portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/pages/users/add.js

```diff
@@ -10,28 +10,59 @@ function onPost(env) {
     var domainNames = getDomainNames(env);
     var primaryDomainName = getPrimaryDomainName(env);
     sendPasswordStrengthParameters();
-    var claimMap = {};
-    var credentialMap = {};
-    var domain = null;
-    credentialMap["password"] = env.request.formParams['newPassword'];
-    claimMap["http://wso2.org/claims/username"] = env.request.formParams['inputUsername'];
-    domain = env.request.formParams['domain'];
-    var registrationResult = userRegistration(claimMap, credentialMap, domain);
-    if (registrationResult.errorMessage) {
+
+    var optionSelector = env.request.formParams['verificationSelector'];
+
+    if (optionSelector === "with_password") {
+        var credentialMap = {};
+        var claimMap = {};
+        var domain = null;
+        credentialMap["password"] = env.request.formParams['newPassword'];
+        claimMap["http://wso2.org/claims/username"] = env.request.formParams['inputUsername'];
+        domain = env.request.formParams['domain'];
+        var registrationResult = userRegistration(claimMap, credentialMap, domain);
+        if (registrationResult.errorMessage) {
+            return {
+                domainNames: domainNames, primaryDomainName: primaryDomainName,
+                errorMessage: registrationResult.errorMessage
+            };
+        }
+        else {
+            return {
+                domainNames: domainNames,
+                primaryDomainName: primaryDomainName,
+                message: registrationResult.message
+            };
+            //TODO:do a redirect to listing page once user added.
+            //sendRedirect(env.contextPath + '/users/list');
+        }
+    }
+    else if (optionSelector === "ask_password") {
+        var credentialMapAP = {};
+        var claimMapAP = {};
+        var domainAP = null;
+        var emailMapAP = {};
+        claimMapAP["http://wso2.org/claims/username"] = env.request.formParams['inputUsername'];
```

Comment:
since the claims,credentials and domain value retrieval is similar to the add user manually, try to make it to a common method. We can reduce the code duplication.